Tella/Scenes/Resources/ViewModel/ResourcesViewModel.swift

```diff
@@ -0,0 +1,218 @@
+                }
+            },
+            receiveValue: { data in
+                completion(.success(data))
+            }
+        )
+        .store(in: &cancellables)
+    }
+    
+    private func saveToVault(data: Data, resource: Resource, serverId: Int) {
+        do {
+            guard let resourceIsSaved = try self.appModel.tellaData?.addResource(resource: resource, serverId: serverId, data: data) else {
+                toggleIsLoadingResource(id: resource.id)
+                return
+            }
+
+            if resourceIsSaved {
+                DispatchQueue.main.async {
+                    self.getDownloadedResources()
+                    self.availableResources.removeAll { $0.id == resource.id }
+                }
+            }
+        } catch {
+            debugLog(error)
+        }
+    }
+
+    
+    func getDownloadedResources() {
+        self.downloadedResources = fetchDownloadedResources()
+    }
+    
+    func fetchDownloadedResources() -> [DownloadedResourcesList] {
+        guard let resources = self.appModel.tellaData?.getResources() else {
+            return []
+        }
+
+        return resources.map { resource in
+            return DownloadedResourcesList(
+                id: resource.id,
+                externalId: resource.externalId,
+                resourceCard: ResourceCardViewModel(
+                    title: resource.title,
+                    serverName: resource.server?.name ?? "",
+                    type: .more,
+                    action: { self.showResourceBottomSheet(resourceId: resource.id, resourceTitle: resource.title)}
+                ),
+                fileName: resource.fileName,
+                size: resource.size,
+                createdAt: resource.createdAt
+            )
+        }
+    }
+    
+    func deleteResource(resourceId: String) -> Void {
+        let result = self.appModel.tellaData?.deleteDownloadedResource(resourceId: resourceId)
+        switch result {
+        case .success(_):
+            self.getDownloadedResources()
+            self.getAvailableForDownloadResources()
+        case .failure(let error):
+            debugLog(error)
+            Toast.displayToast(message: error.localizedDescription)
+        default:
+            break
+        }
+    }
+    
+    func openResource(resourceId: String, fileName: String) -> URL? {
+        guard let url = self.appModel.vaultManager.loadFileToURL(fileName: fileName, fileExtension: "pdf", identifier: resourceId) else { return nil }
+        return url
+    }
+    
+    private func toggleIsLoadingResource(id: String) {
+        if let index = self.availableResources.firstIndex(where: { $0.id == id }) {
+            self.availableResources[index].isLoading = !self.availableResources[index].isLoading
+            self.availableResources = self.availableResources.map { $0 }
+        }
+    }
+    
+    private func showResourceBottomSheet(resourceId: String, resourceTitle: String) {
```

Comment:
I suggest to use `resource` instead of  `resourceId` and `resourceTitle` and use  `selectedResource` 

    private func showResourceBottomSheet(resource:Resource) {
        self.selectedResource = resource
        self.onShowResourceBottomSheet?()
    }

Reply:
we will only use selectedResource for this function. Why add a new variable instead of passing the specific values that we need for showResourceBottomSheet?

Reply:
The `selectedResource` var  will be used by `showResourceBottomSheet`, `showDeleteResourceConfirmationView`